### PR TITLE
CONCF-509 Return the main page data OR the article data

### PR DESCRIPTION
**Pull request by rogatty**
https://wikia-inc.atlassian.net/browse/CONCF-509

/cc @bognix @Warkot @tomasznapieralski

### CODE REVIEW

**Comment by bognix** [comment]
LGTM

File: extensions/wikia/MercuryApi/MercuryApiController.class.php [hunk 1]
```diff
@@ -77,12 +77,46 @@ private function getArticleDetails( $articleId ) {
...
+	 * @throws WikiaException
+	 */
+	private function getArticleDescription( $articleId, $descLength = 100 ) {
+		$article = Article::newFromID( $articleId );
+		$title = $article->getTitle();
+		$sMessage = null;
+
+		if ( $title->isMainPage() ) {
+			// we're on Main Page, check MediaWiki:Description message
+			$sMessage = wfMessage( 'Description' )->text();
```
Comment (Warkot) [inline extensions/wikia/MercuryApi/MercuryApiController.class.php]: Is it safe to use `->text()`?

**Comment by Warkot** [comment]
If you're sure that moving the method doesn't break anything, then LGTM. Otherwise, why not just make it public and reference it?

File: extensions/wikia/MercuryApi/MercuryApiController.class.php (hunk 1 above)
Comment (rogatty) [inline extensions/wikia/MercuryApi/MercuryApiController.class.php]: It's copied from https://github.com/Wikia/app/blob/dev/extensions/wikia/ArticleMetaDescription/ArticleMetaDescription.php#L64

But from what I see here: https://github.com/Wikia/app/blob/dev/includes/GlobalFunctions.php#L1358
`wfMessage` always returns instance of `Message`. Should be safe.

**Comment by rogatty** [comment]
`getArticleDescription` was written (mostly copied from `ArticleMetaDescription` extension ;)) by me and used only for this case. I wouldn't like to spend too much time on thinking where to put it as a public method before it's actually needed elsewhere.

**Comment by Warkot** [comment]
LGTM then. (y)

**Comment by tomasznapieralski** [comment]
LGTM

**Pull request merged**